### PR TITLE
#162725221 Implement social login for google and facebook

### PR DESCRIPTION
**Pull request by MuhweziDeo**
#### What does this PR do?
Enables social login for google and facebook

#### Description of Task to be completed?
- create a presentational component for social login buttons
- create a container component to handle state for social authentication
- connect to facebook and twitter apps for Xmen
- set up actions, action Types and reducers for social authentication
- connect to facebook and google to get access tokens and post to the  backend
- register users successfully and store tokens in local storage

#### How should this be manually tested?
- Fetch and checkout into this branch with `git fetch origin ft-social-authentication-162725221 && git checkout ft-social-authentication-162725221`
- Install dependencies and start the app with `yarn install && yarn start`
- navigate to `https://localhost:8080/login` and click on one of the icons you which to use for login
- you will be redirected to a login page for one the platforms
- login with your credentials and if it's successful you will be redirected to the home page
- you might be faced with a problem of `https` because facebook login can only be accessed via an `https` connection just add exceptions for all your browsers
- To run tests run the command `yarn run test:coverage`

#### What are the relevant pivotal tracker stories?
[#162725221](https://www.pivotaltracker.com/story/show/162725221)

[Starts #162725221]

**Screenshots**
screen shot 2019-02-17 at 6 01 24 pm
screen shot 2019-02-17 at 6 08 57 pm

### CODE REVIEW

File: src/views/socialAuthView/socialAuthView.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    handlegoogleResponseFailure=(response) => {
+      toastr.warning(response.error);
+      const { googleSignInAction } = this.props;
+      googleSignInAction('invalid request');
+    }
+
+    componentWillReceiveProps(nextprops) {
+      if (nextprops.socialAuthState.isAuthenticated) {
+        toastr.info('Login successful');
```
Comment (malfahad) [inline src/views/socialAuthView/socialAuthView.js]: @MuhweziDeo Please consider  adding the username of the logged in user to the success message.

Comment (MuhweziDeo) [reply src/views/socialAuthView/socialAuthView.js]: cool

File: src/reducers/socialAuthReducer/socialAuthReducer.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        payload:action.payload
+
+      };
+    case GOOGLE_SIGNIN_FAIL:
+      return {
+        ...state,
+        payload: action.payload,
+      };
+    case FACEBOOK_SIGNIN_SUCCESS:
+    console.log(action.payload)
```
Comment (RrNn) [inline src/reducers/socialAuthReducer/socialAuthReducer.js]: Remove the `console.log`

File: src/actions/socialActions/__tests__/socialActions.test.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const middlewares = [thunk];
+const mockStore = configureMockStore(middlewares);
+
+
+describe('async actions', () => {
+  afterEach(() => {
+    fetchMock.restore();
+  });
+
+  it('should create an action for google login fail', () => {
```
Comment (RrNn) [inline src/actions/socialActions/__tests__/socialActions.test.js]: This `it` block doesn't have any assertions and expectations. Delete it.

**Review by RrNn — DISMISSED** [review]
Hi @MuhweziDeo, please move the `socialAuth.css` file to the `assets` folder, remember to add a new blank line at the end too.
Lets store our BASE_URL in the environment. Douglas already has a setup for it.

In the `components/socialAuth/socialAuth.js` file, move the `FaceBook appId` and `GoogleClietId` to the `.env` file. These are a big secret.

**Review by malfahad — DISMISSED** [review]
functionality works.

**Review by armstrongsouljah — COMMENTED** [review]
Hey @MuhweziDeo. I noticed you have `hound violations` maybe first fix those.

File: src/views/socialAuthView/socialAuthView.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  componentWillReceiveProps(nextprops) {
+    if (nextprops.socialAuthState.isAuthenticated) {
+      localStorage.setItem('token', nextprops.socialAuthState.token);
+      localStorage.setItem('username', nextprops.socialAuthState.payload.user.username);
+      setTimeout(() => {
+        this.props.history.push('/');
+      }, 800);
+      toast.success(`Successully Logged in as ${nextprops.socialAuthState.payload.user.username}`);
+    } else {
+      toast.error('An error occurred please try again letter');
```
Comment (RrNn) [inline src/views/socialAuthView/socialAuthView.js]: An error occurred please try again **later**

**Pull request merged**